Tests/TSCBasicTests/ProcessTests.swift

```diff
@@ -42,20 +42,57 @@ class ProcessTests: XCTestCase {
     }
 
     func testPopen() throws {
+        #if os(Windows)
+        let echo = "echo.exe"
+        #else
+        let echo = "echo"
+        #endif
         // Test basic echo.
-        XCTAssertEqual(try Process.popen(arguments: ["echo", "hello"]).utf8Output(), "hello\n")
+        XCTAssertEqual(try Process.popen(arguments: [echo, "hello"]).utf8Output(), "hello\n")
 
         // Test buffer larger than that allocated.
         try withTemporaryFile { file in
             let count = 10_000
             let stream = BufferedOutputByteStream()
             stream <<< Format.asRepeating(string: "a", count: count)
             try localFileSystem.writeFileContents(file.path, bytes: stream.bytes)
-            let outputCount = try Process.popen(args: "cat", file.path.pathString).utf8Output().count
+            #if os(Windows)
+            let cat = "cat.exe"
+            #else
+            let cat = "cat"
+            #endif
+            let outputCount = try Process.popen(args: cat, file.path.pathString).utf8Output().count
             XCTAssert(outputCount == count)
         }
     }
 
+    func testPopenAsync() throws {
+        #if os(Windows)
+        let args = ["where.exe", "where"]
+        let answer = "C:\\Windows\\System32\\where.exe"
```

Comment:
There is no guarantee that the path is `C:\Windows\System32\where.exe` . The correct answer is `%SystemRoot%\System32\where.exe`.  You can use `GetSystemDirectoryW` to get the value for `%SystemRoot%` rather than expanding it via an environment variable.